### PR TITLE
Bug 1823999 - Fix UNDO Snackbar for composified Tabs Tray

### PR DESCRIPTION
**Pull request by MozillaNoah**
https://user-images.githubusercontent.com/87384386/234960614-1a8a5123-58a6-4846-99f5-111ab0d635d2.mov

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-an

droid/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1823999

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFab.kt [hunk 1]
```diff
@@ -81,22 +74,14 @@ fun TabsTrayFab(
...
-                    .padding(16.dp),
-                contentDescription = contentDescription,
-                label = label,
-                onClick = onClick,
-            )
-        }
+    if (isInNormalMode) {
+        FloatingActionButton(
+            icon = icon,
+            modifier = Modifier.padding(16.dp),
```
Comment (Alexandru2909) [inline fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFab.kt:80]: This is a misstep from my PR, but could we also change this modifier to apply padding only for `bottom` and `end`? this way we lower the position of the `Undo` snackbar and achieve parity with the xml version.

**Review by Alexandru2909 — APPROVED** [review]
I left a comment about the snackbar position compared to the XML version, but otherwise LGTM!

**Review by MatthewTighe — APPROVED** [review]
LGTM outside of the note from Alexandru

File: fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFab.kt (hunk 1 above)
Comment (MozillaNoah) [reply fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFab.kt:80]: For sure! Thanks for catching this!

File: fenix/app/src/main/res/layout/component_tabstray3_fab.xml [hunk 2]
```diff
@@ -3,6 +3,7 @@
    - License, v. 2.0. If a copy of the MPL was not distributed with this
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
 <androidx.compose.ui.platform.ComposeView xmlns:android="http://schemas.android.com/apk/res/android"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:layout_gravity="bottom|end"
```
[the remark this answers is not in the document]
Comment (MozillaNoah) [reply fenix/app/src/main/res/layout/component_tabstray3_fab.xml:8]: It is. The change here is to force the FAB into the bottom right corner of the `TabsTrayDialogFragment`. The changes he commented above are to reduce the spacing between Snackbars and the FAB.

**Pull request merged**